### PR TITLE
Feature: Support changing behavior of buttons with modifier keys

### LINKED ISSUE #12117: Feature: Change the behavior of toolbar and flyout buttons with modifier keys
### What feature or improvement do you think would benefit Files?

The ability to change the behavior of toolbar and flyout buttons with modifier keys would allow for a greater variety of operations.
For example, In FIle Explorer the delete button acts as permanent deletion while pressing `Shift`, but it doesn't in Files.

### Requirements

Create a mechanism to change the behavior of toolbar and flyout buttons with modifier keys.

### Files Version

2.4.65.0

### Windows Version

Windows 11 22H2

### Comments

_No response_

### PR DESCRIPTION
**Pull request by hishitetsu**
**Resolved / Related Issues**
- [X] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #12117 

**Validation**
How did you test these changes?
- [X] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [X] Are there any other steps that were used to validate these changes?
   1. Holding down Shift and pressing the delete button on the toolbar or flyout will result in permanent deletion.
   2. Holding down Shift and pressing the paste button on the toolbar will paste items into selected folder. If no items are selected, it will act as normal pasting.

### CODE REVIEW

File: src/Files.App/Commands/Manager/CommandManager.cs [hunk 1]
```diff
@@ -168,6 +174,14 @@ public CommandManager()
...
 				.ToImmutableDictionary(command => command.Code);
 
+			_ModifiableDeleteItem = new ModifiableCommand(commands[CommandCodes.DeleteItem], new() {
+				{ KeyModifiers.Shift,  commands[CommandCodes.DeleteItemPermanently] }
+			});
+
+			_ModifiablePasteItem = new ModifiableCommand(commands[CommandCodes.PasteItem], new() {
+				{ KeyModifiers.Shift,  commands[CommandCodes.PasteItemToSelection] }
+			});
+
```
Comment (QuaintMako) [inline src/Files.App/Commands/Manager/CommandManager.cs:184]: I am not a fan at all of instanciating those methods here. They should at least be done inside of a dedicated method, possibly called by the constructor. I'd go as far as create a smaller class contained by the Command Manager to store the modifiable commands? If we get more modifiables here, it's gonna be really crowded soon.

File: src/Files.App/Commands/Manager/CommandManager.cs [hunk 2]
```diff
@@ -563,5 +577,112 @@ private void Action_PropertyChanged(object? sender, PropertyChangedEventArgs e)
 				}
 			}
 		}
+
+		[DebuggerDisplay("Command {Code} (Modifiable)")]
+		private class ModifiableCommand : ObservableObject, IRichCommand
```
Comment (QuaintMako) [inline src/Files.App/Commands/Manager/CommandManager.cs:582]: To move to a new file. The concept is too important to be shove under another class.

File: src/Files.App/Commands/Manager/CommandManager.cs [hunk 3]
```diff
@@ -563,5 +577,112 @@ private void Action_PropertyChanged(object? sender, PropertyChangedEventArgs e)
...
+			}
+
+			public bool IsExecutable => BaseCommand.IsExecutable;
+
+			public ModifiableCommand(IRichCommand baseCommand, Dictionary<KeyModifiers, IRichCommand> modifiedCommands)
+			{
+				BaseCommand = baseCommand;
+				ModifiedCommands = modifiedCommands.ToImmutableDictionary();
+
+				if (baseCommand is ActionCommand actionCommand)
```
Comment (QuaintMako) [inline src/Files.App/Commands/Manager/CommandManager.cs:626]: Invert if to reduce nesting.

**Review by QuaintMako — COMMENTED** [review]
Some feedback about the code. I cannot test it though. Some interesting feature that might be very powerful on the long run.

To be honest fully, there is a little something feeling wrong to me but I cannot pinpoint what... I'll be sure to come back to elaborate when I have IDE to review it properly.

File: src/Files.App/Commands/Manager/CommandManager.cs (hunk 1 above)
Comment (hishitetsu) [reply src/Files.App/Commands/Manager/CommandManager.cs]: The instantiating of commands has been carved out into a separate method. Also, the management of ModifiableCommand itself has been carved out into a new class.

File: src/Files.App/Commands/Manager/CommandManager.cs (hunk 3 above)
Comment (hishitetsu) [reply src/Files.App/Commands/Manager/CommandManager.cs]: Since this code also serves as a cast, I left it as is.

**Review by QuaintMako — APPROVED** [review]
I cannot test the code, yet with the suggested refactoring done, I trust the concept way more!

LGTM, I'll need someone else to test the code.

**Pull request merged**